grammar.js

```diff
@@ -1079,16 +1080,35 @@ module.exports = grammar({
       '=',
       field('value', $._expression),
       field('consequence', $.block),
-      optional(field('alternative', $.else_clause))
+      repeat($._else_clause),
+    ),
+
+    _else_clause: $ => choice(
```

Comment:
This complicates the grammar IMHO, which is a non-goal of the project so I don't think
we should merge this.

Reply:
Since I filed this, let-else has come very close to stabilisation. I did some work on it. Don't you need an "else clause" rule for that anyway?

Reply:
Having a flatter tree structure benefits editor plugins (as demonstrated in the OP) which is actually a goal of the project. Keeping the grammar simple is desirable, but not a goal as far as I'm aware.

In any case I don't see how this complicates the grammar. If anything it's much simpler to consume the `if` node this way instead of recursively nesting it like a matryoshka doll.

Needless to say I disagree with closing this.

Reply:
Keeping the grammar simple is one of the goal of tree-sitter grammars. I've seen a few PRs that were restricting the grammar to not accept invalid programs to actually be refused, because they were making the grammar more complex.

Reply:
I agree that this PR is desirable, even if it increases grammar complexity by a small amount. We use the parsed tree to compute indentation levels and this nested `else` clause completely trips up our algorithm.

Reply:
I've gone through closed PRs, there's only a single instance where a PR was closed, but that's because a simpler approach existed. https://github.com/tree-sitter/tree-sitter-rust/pull/70#issuecomment-623626465

Reply:
Could we address the concern by removing the `else_if_let_clause` in favour of a generic `else_if` containing a more generic let expression? The extra clause type is a poor choice on my part, and thinking of a refutable let statement as a boolean expression with name-binding abilities is actually how rustc thinks about it anyway.

Reply:
The only caveat is that if-let doesn't currently accept a type annotation, which is probably why I opted not to go that route originally. (The RFC'd let-else syntax does!) But according to the discussion here, accepting more than what is technically Rust syntax sounds acceptable.